RULES/cc/cc.go

```diff
@@ -214,23 +214,43 @@ func (lib Library) arRule() core.BuildRule {
 	// There is no option to ar to always force creation of a new archive; the "c"
 	// modifier simply suppresses a warning if the archive doesn't already
 	// exist. So instead we delete the target (out) if it already exists.
-	return core.BuildRule {
-		Name: toolchain.Name() + "-ar",
-		Variables: map[string]string{
-			"command": fmt.Sprintf("rm -f $out 2> /dev/null; %s rcs $out $in", ninjaEscape(toolchain.Archiver())),
-			"description": fmt.Sprintf("AR (toolchain: %s) $out", toolchain.Name()),
-		},
+	if strings.HasSuffix(toolchain.Name(), "efi") {
```

Comment:
I don't really like relying on the toolchain name if it can be avoided. Have you considered creating an interface (something like `bool isEfiToolchain()`) or move the generation of build rules to the toolchain and have each toolchain implement `arRule()`, `soRule()` etc. however it likes? How will these new EFI toolchains look? Will this be standard clang-14 toolchain just with a different name?